src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/GetReservation.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.Reservations.Common;
+using Microsoft.Azure.Management.Reservations.Models;
+using Microsoft.Azure.Commands.Reservations.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Reservations;
+
+namespace Microsoft.Azure.Commands.Reservations.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmReservation", DefaultParameterSetName = Constants.ParameterSetNames.CommandParameterSet), OutputType(typeof(PSReservationPage), typeof(PSReservation))]
+    public class GetReservation : AzureReservationsCmdletBase
+    {
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNull]
+        public string ReservationOrderId { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        public string ReservationId { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.ObjectParameterSet,
+            Mandatory = false,
+            ValueFromPipeline = true)]
+        [ValidateNotNull]
+        public PSReservationOrder ReservationOrder { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.ObjectParameterSet,
```

Comment:
You may only have one object take its value from the pipeline for each parameter set.  If these both can take their value from the pipeline, they must be in separate parameter sets.